pkg/api/validation/validation.go

```diff
@@ -2562,6 +2572,17 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("activeDeadlineSeconds"), newPod.Spec.ActiveDeadlineSeconds, "must not update from a positive integer to nil value"))
 	}
 
+	// validate updated spec.PriorityClassName.
+	if newPod.Spec.PriorityClassName != oldPod.Spec.PriorityClassName && len(newPod.Spec.PriorityClassName) > 0 {
```

Comment:
No, I am not disallowing "foo" -> "bar". That condition just checks that if there is a change, then it validates the new name. It doesn't prevent the change of priority class name.